alacritty/src/display/mod.rs

```diff
@@ -552,9 +558,59 @@ impl Display {
         }
     }
 
-    pub fn make_current(&self) {
-        if !self.context.get().is_current() {
-            self.context.make_current(&self.surface).expect("failed to make context current")
+    pub fn make_current(&mut self, recover: bool) {
+        let is_current = self.context.get().is_current();
+        if is_current && !recover {
+            return;
+        }
+
+        let mut was_context_reset = if is_current {
+            false
+        } else {
+            match self.context.make_current(&self.surface) {
+                Err(err) if err.error_kind() == ErrorKind::ContextLost => {
+                    info!("Context lost for window {:?}", self.window.id());
+                    true
+                },
+                _ => false,
+            }
+        };
```

Comment:
```suggestion
        // Attempt to make the context current, if it is not.
        let mut was_context_reset = if is_current {
            false
        } else {
            match self.context.make_current(&self.surface) {
                Err(err) if err.error_kind() == ErrorKind::ContextLost => {
                    info!("Context lost for window {:?}", self.window.id());
                    true
                },
                _ => false,
            }
        };
```

I think a comment here is nice to explain that our goal here is not just checking if the context was reset, but actually performing the make_current call.